x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/Expressions.java

```diff
@@ -93,8 +93,29 @@ public static boolean match(List<? extends Expression> exps, Predicate<? super E
         return false;
     }
 
+    /**
+     * Return the logical AND of a list of {@code Nullability}
+     * <pre>
+     *  UNKNOWN AND TRUE/FALSE/UNKNOWN = UNKNOWN
+     *  FALSE AND FALSE = FALSE
+     *  TRUE AND FALSE/TRUE = TRUE
+     * </pre>
+     */
     public static Nullability nullable(List<? extends Expression> exps) {
```

Comment:
Moved this code here since I found the existing loop quite inefficient - make a stream to create extraction to put into an array to pass it to a method that does another looping to do basic checks...